kubetest2-gke/deployer/version.go

```diff
@@ -0,0 +1,169 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package deployer
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"google.golang.org/api/container/v1"
+	"gopkg.in/yaml.v2"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"sigs.k8s.io/kubetest2/pkg/exec"
+)
+
+var (
+	noneReleaseChannel    = "None"
+	rapidReleaseChannel   = "rapid"
+	regularReleaseChannel = "regular"
+	stableReleaseChannel  = "stable"
+
+	validReleaseChannels = sets.NewString(noneReleaseChannel, rapidReleaseChannel, regularReleaseChannel, stableReleaseChannel)
+)
+
+func validateVersion(version string) error {
+	switch version {
+	case "latest", "":
+		return nil
+	default:
+		re, err := regexp.Compile(`(\d)\.(\d)+(\.(\d)*(.*))?`)
+		if err != nil {
+			return err
+		}
+		if !re.MatchString(version) {
+			return fmt.Errorf("unknown version %q", version)
+		}
+	}
+	return nil
+}
+
+func validateReleaseChannel(releaseChannel string) error {
+	if releaseChannel != "" && !validReleaseChannels.Has(releaseChannel) {
```

Comment:
maybe can reuse `toReleaseChannel` to avoid needing sets.

Reply:
`toReleaseChannel` is used to convert the channel name in the server config to the channel name in the `gcloud` flag, so it cannot be used here. I have changed to simple array iteration to avoid using sets though.

Reply:
I meant since the strings are the same, we can use `toLower`/`toUpper` and the default case to also use it as a validation method.
But considering it's just 4 values the array solution is fine as well.